scipy/stats/_discrete_distns.py

```diff
@@ -514,6 +526,144 @@ def _logcdf(self, k, M, n, N):
+                                   {{N \choose K}}
+
+    for :math:`k \in [0, K]`, :math:`K \in [0, N]`, :math:`r \in [0, N-K]`,
+    and the binomial coefficient is:
+
+    .. math:: \binom{n}{k} \equiv \frac{n!}{k! (n - k)!}.
+
+    It is equivalent to observing :math:`k` successes in :math:`k+r-1`
+    samples with :math:`k+r`'th sample being a failure. The former
+    can be modelled as a hypergeometric distribution. The probability
+    of the latter is simply the number of failures remaining
+    :math:`N-K-(r-1)` divided by the size of the remaining population
+    :math:`N-(k+r-1)`. This relationship can be shown as:
+
+    .. math:: NHG(k;N,K,r) = HG(k;N,K,k+r-1)\frac{(N-K-(r-1))}{(N-(k+r-1))}
+
+    where :math:`NHG` is probability mass function (PMF) of the
+    negative hypergeometric distribution and :math:`HG` is the
+    PMF of the hypergeometric distribution.
+
+    %(after_notes)s
+
+    Examples
+    --------
+    >>> from scipy.stats import nhypergeom
+    >>> import matplotlib.pyplot as plt
+
+    Suppose we have a collection of 20 animals, of which 7 are dogs.
+    Then if we want to know the probability of finding a given number
+    of dogs (successes) in a sample with exactly 12 animals that
+    aren't dogs (failures), we can initialize a frozen distribution
+    and plot the probability mass function:
+
+    >>> N, K, r = [20, 7, 12]
+    >>> rv = nhypergeom(N, K, r)
+    >>> x = np.arange(0, K+2)
+    >>> pmf_dogs = rv.pmf(x)
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> ax.plot(x, pmf_dogs, 'bo')
+    >>> ax.vlines(x, 0, pmf_dogs, lw=2)
+    >>> ax.set_xlabel('# of dogs in our group with given 12 failures')
+    >>> ax.set_ylabel('nhypergeom PMF')
+    >>> plt.show()
+
+    Instead of using a frozen distribution we can also use `nhypergeom`
+    methods directly.  To for example obtain the probability mass
+    function, use:
+
+    >>> prb = nhypergeom.pmf(x, N, K, r)
+
+    And to generate random numbers:
+
+    >>> R = nhypergeom.rvs(N, K, r, size=10)
+
+    To verify the relationship between `hypergeom` and `nhypergeom`, use:
+
+    >>> from scipy.stats import hypergeom, nhypergeom
+    >>> N, K, r = 45, 13, 8
+    >>> k = 6
+    >>> nhypergeom.pmf(k, N, K, r)
+    0.06180776620271643
+    >>> hypergeom.pmf(k, N, K, k+r-1) * (N - K - (r-1)) / (N - (k+r-1))
+    0.06180776620271644
+
+    See Also
+    --------
+    hypergeom, binom, nbinom
+
+    References
+    ----------
+    .. [1] Negative Hypergeometric Distribution on Wikipedia
+           https://en.wikipedia.org/wiki/Negative_hypergeometric_distribution
+
+    .. [2] Negative Hypergeometric Distribution from
+           http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Negativehypergeometric.pdf
+
+    """
+    def _get_support(self, N, K, r):
+        return 0, K
+
+    def _argcheck(self, N, K, r):
+        cond = (K >= 0) & (K <= N)
+        cond &= (r >= 0) & (r <= N-K)
```

Comment:
This must be computed in a single expression for broadcasting of all the arguments to work correctly:
```suggestion
        cond = (K >= 0) & (K <= N) & (r >= 0) & (r <= N-K)
```
Without that change, we get the following error:
```
In [2]: nhypergeom.pmf(0, [[10], [20]], 6, [0, 3])                                                                                                         
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-0934ec12a9d9> in <module>
----> 1 nhypergeom.pmf(0, [[10], [20]], 6, [0, 3])

~/mc38/lib/python3.8/site-packages/scipy-1.6.0.dev0+371e21f-py3.8-macosx-10.9-x86_64.egg/scipy/stats/_distn_infrastructure.py in pmf(self, k, *args, **kwds)
   3023         _a, _b = self._get_support(*args)
   3024         k = asarray((k-loc))
-> 3025         cond0 = self._argcheck(*args)
   3026         cond1 = (k >= _a) & (k <= _b) & self._nonzero(k, *args)
   3027         cond = cond0 & cond1

~/mc38/lib/python3.8/site-packages/scipy-1.6.0.dev0+371e21f-py3.8-macosx-10.9-x86_64.egg/scipy/stats/_discrete_distns.py in _argcheck(self, N, K, r)
    631     def _argcheck(self, N, K, r):
    632         cond = (K >= 0) & (K <= N)
--> 633         cond &= (r >= 0) & (r <= N-K)
    634         return cond
    635 

ValueError: non-broadcastable output operand with shape (2,1) doesn't match the broadcast shape (2,2)
```

Reply:
True. Inplace operations fail to broadcast shapes. I have changed it now